sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -3465,10 +3534,12 @@ class AstBuilder extends SqlBaseParserBaseVisitor[AnyRef] with SQLConfHelper wit
       properties += PROP_LOCATION -> _
     }
 
-    CreateNamespace(
-      UnresolvedNamespace(visitMultipartIdentifier(ctx.multipartIdentifier)),
-      ctx.EXISTS != null,
-      properties)
+    withIdentClause(ctx.identifierReference, ident => {
```

Comment:
Sorry to suggest this at the last minute, but I feel the way we handle MERGE looks pretty neat. How about we change this to
```
CreateNamespace(
  withIdentClause(ctx.identifierReference, UnresolvedNamespace(_)),
  ctx.EXISTS != null,
  properties
)
```

The code diff will be much smaller because it is just a one-line change.